Lib/test/test_tarfile.py

```diff
@@ -2498,6 +2498,70 @@ def test_partial_input_bz2(self):
         self._test_partial_input("r:bz2")
 
 
+class SymlinkOverwriteTest(unittest.TestCase):
+    # The testcase checks for correct overwriting of an
+    # existing symlink (issue #12800)
+
+    @support.skip_unless_symlink
+    def test_overwrite_symlink(self):
+        tmpdir = support.temp_cwd('overwrite_symlink')
```

Comment:
temp_cwd is a context manager, please use it in a with statement (also you're not using the variable at all which is strange).